cfg_mgmt/util.py

```diff
@@ -297,7 +301,31 @@ def rotate_config_element_and_persist_in_cfg_repo(
         git_helper.add_and_commit(
             message=commit_message,
         )
-        git_helper.push('@', target_ref)
+        for attempt in range(1, MAX_RETRIES + 1):
+            try:
+                git_helper.push('@', target_ref)
+                logger.info(f'Successfully pushed changes to GitHub on attempt {attempt}')
+                break
+            except Exception as e:
+                if attempt < MAX_RETRIES:
+                    logger.warning(f'GitHub push failed (attempt {attempt}/{MAX_RETRIES}): {e}')
+                    logger.info(f'Retrying in {RETRY_DELAY} seconds...')
+
+                    # pull the latest changes and rebase
+                    latest_commit = git_helper.fetch_head(target_ref)
```

Comment:
chance for success will be greater if we do fetch + hard-reset + recreate changes